frontend/connection/src/constants.ts

```diff
@@ -47,10 +47,26 @@ export const PING_MAXIMUM_RETRY_PERIOD_MS = 2000
  */
 export const MAX_RETRIES_BEFORE_CLIENT_ERROR = 6
 
+/**
+ * Detect if we're on an Android device. This is used to adjust timeouts
+ * for Android Chrome where file pickers can keep the app backgrounded
+ * for extended periods.
+ */
+const isAndroidDevice = (): boolean => {
+  if (typeof navigator === "undefined") {
+    return false
+  }
+  return /Android/i.test(navigator.userAgent)
+}
+
 /**
  * Timeout when attempting to connect to a websocket, in millis.
+ * Android devices get a longer timeout (60s vs 15s) because file pickers
+ * background the browser tab for extended periods, causing premature
+ * connection timeouts.
+ * See: https://github.com/streamlit/streamlit/issues/11419
  */
-export const WEBSOCKET_TIMEOUT_MS = 15 * 1000
+export const WEBSOCKET_TIMEOUT_MS = isAndroidDevice() ? 60 * 1000 : 15 * 1000
```

Comment:
60 seconds should be a lot less annoying for uploading files compared to 15 seconds timeout.